app/services/storage-monitor.js

```diff
@@ -0,0 +1,55 @@
+import Service, { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+import mdCalcStorage from 'mdeditor/utils/md-calc-storage';
+
+export default Service.extend({
+  init() {
+    this._super(...arguments);
+
+    this.storageMessages();
```

Comment:
Unless I'm missing something, the message is only produced when the service is initialized? It needs to perform the check **_before_** every attempted update to localstorage, otherwise the user can save data that will exceed the quota which causes an immediate error (it appears that the update fails and the app reboots to the previous state, but any data from the update would be lost).